pkg/core/native/ledger.go

```diff
@@ -142,6 +149,19 @@ func (l *Ledger) getTransactionFromBlock(ic *interop.Context, params []stackitem
 	return TransactionToStackItem(block.Transactions[index])
 }
 
+// getTransactionVMState returns VM state got after transaction invocation.
+func (l *Ledger) getTransactionVMState(ic *interop.Context, params []stackitem.Item) stackitem.Item {
+	tx, h, err := getTransactionAndHeight(ic.DAO, params[0])
+	if err != nil || !isTraceableBlock(ic.Chain, h) {
+		return stackitem.Make(vm.NoneState)
+	}
+	aer, err := ic.DAO.GetAppExecResults(tx.Hash(), trigger.Application)
```

Comment:
Yes, that's because `(dao *Simple) GetAppExecResults` doesn't return the transaction height. The problem is that `(dao *Simple) GetAppExecResults` can return both block and transaction+height, so do we need to split it into two separate methods?

Or maybe it's even better to create some `(dao *Simple) GetTxExecResult (uint32, *transaction.Transaction, state.AppExecResult)` method?

Reply:
I had an idea for creating it way back when I'd changed the DB format, but there was no real use case. Now it looks like we have it. There are cases where we only need transaction/height and we don't want to spend time on execution result decoding, but if we have to decode everything then it's worth returning everything from some method.

Reply:
Fixed. Let's first come up to the final decision about `GetTxExecResult`, and after that I'll update gomods.